sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -173,11 +177,13 @@ def read(  # pylint:disable=docstring-missing-param
             request_options["populatePartitionKeyRangeStatistics"] = populate_partition_key_range_statistics
         if populate_quota_info is not None:
             request_options["populateQuotaInfo"] = populate_quota_info
-        collection_link = self.container_link
-        self._properties = self.client_connection.ReadContainer(
-            collection_link, options=request_options, **kwargs
-        )
-        return self._properties
+        container = self.client_connection.ReadContainer(self.container_link, options=request_options, **kwargs)
+        # Only cache Container Properties that will not change in the lifetime of the container
+        self.client_connection.container_properties_cache[self.container_link] = {
```

Comment:
`self.container_link` contains the name? If so, how is this PR addressing the recreation scenario? Or is that being done in a separate PR?

Reply:
That is being done in a separate pr I am working on. That one should be out soon after this PR is merged.